services/_shared/llm_client.py

```
@@ -878,28 +927,40 @@ def _call_backend(
-            resp.status_code in _RETRYABLE_STATUSES
-            and attempt < config.retry_attempts - 1
-        ):
-            _RETRY_SLEEP(_RETRY_BASE_DELAY * (2 ** attempt))
+            result_q.put(("ok", resp))
+        except Exception as e:  # noqa: BLE001 - re-raised on the waiting side
+            result_q.put(("error", e))
+
+    threading.Thread(target=_do_call, daemon=True).start()
+    while True:
+        if cancel_event.is_set():
+            raise httpx.RequestError("cancelled mid-flight")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)